loader.py

```diff
@@ -0,0 +1,349 @@
+import chess
+import bz2
+import torch
+import torch.multiprocessing as mp
+from collections import deque
+import numpy as np
+import random
+import os
+from urllib.request import urlopen
+
+
+# initial setup that looks ugly
+FIRST_CORDS = {'a': 0, 'b': 1, 'c': 2, 'd': 3, 'e': 4, 'f': 5, 'g': 6, 'h': 7}
+SECOND_CORDS = {'1': 0, '2': 1, '3': 2, '4': 3, '5': 4, '6': 5, '7': 6, '8': 7}
+PIECES = {'P': 0, 'N': 1, 'B': 2, 'R': 3, 'Q': 4, 'K': 5,
+          'p': 6, 'n': 7, 'b': 8, 'r': 9, 'q': 10, 'k': 11}
+
+
+class FENGenerator:
+    def __init__(self, replay_queue: mp.Queue, pgn_path):
+
+        """
+        An iterable class that takes game saves and converts them into a chessboard position record in fen notation.
+        :param replay_queue: target queue for samples
+        :param pgn_paths: list of paths to data files
+        """
+
+        self.replay_queue = replay_queue
+        self.error_tolerance=2
+        self.current_path = (
+            0  # index of the path to the currently used data file in the pgn_paths list
+        )
+        self.current_line = (
+            []
+        )  # currently used line from the data file, representing one game
+        self.board = chess.Board()  # the chessboard on which the moves are made
+        if isinstance(pgn_path,str):
+            self.pgn_paths = [pgn_path]
+        else:
+            self.pgn_paths = pgn_path
+        # pgn_file - currently used data file
+        self.pgn_file = None
+        self.get_new_file(0)
+
+    def closeFile(self):

```

Comment:
inconsistent use of camelCase and snake_case - if everywhere else snake_case is used, I'd stay with that